src/Dialog/DialogBottomSheet.tsx

```diff
@@ -0,0 +1,327 @@
+import React, {useEffect, useRef, useState, PropsWithChildren, MouseEvent, TouchEvent, ChangeEvent} from 'react'
+import styled from 'styled-components'
+import {SxProp} from '../sx'
+import {useRefObjectAsForwardedRef} from '../hooks/useRefObjectAsForwardedRef'
+import Box from '../Box'
+import {get} from '../constants'
+
+const ANIMATION_DURATION = 300
+const FULL_HEIGHT = 95
+const HALF_HEIGHT = 50
+
+/**
+ * Props to customize the rendering of the Dialog.
+ */
+export interface DialogBottomSheetProps extends SxProp {
+  /**
+   * This method is invoked when a gesture to close the dialog is used
+   */
+  onClose: (gesture: 'close-button' | 'escape' | 'drag' | 'overlay') => void
+
+  /**
+   * Passed through to use as the draggable area of the dialog.
```

Comment:
Non-blocking nitpick: I'd reword this to "Content that appears in the draggable header area." The current description kind of sounds like it lets you change what we render instead of the the blue drag "handle".

Reply:
Makes sense. Fixed.